extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java

```diff
@@ -81,4 +103,80 @@ private void addReflectiveHierarchyClass(DotName className) {
     private void addReflectiveClass(boolean methods, boolean fields, String... className) {
         reflectiveClass.produce(new ReflectiveClassBuildItem(methods, fields, className));
     }
+
+    // Generate a ObjectMapperCustomizer bean that registers each serializer / deserializer with ObjectMapper
+    @BuildStep
+    void generateCustomizer(BuildProducer<GeneratedBeanBuildItem> generatedBeans,
+            List<JacksonModuleBuildItem> jacksonModules) {
+
+        if (jacksonModules.isEmpty()) {
+            return;
+        }
+
+        ClassOutput classOutput = new ClassOutput() {
+            @Override
+            public void write(String name, byte[] data) {
+                generatedBeans.produce(new GeneratedBeanBuildItem(name, data));
+            }
+        };
+
+        try (ClassCreator classCreator = ClassCreator.builder().classOutput(classOutput)
+                .className("io.quarkus.jackson.customizer.RegisterSerializersAndDeserializersCustomizer")
+                .interfaces(ObjectMapperCustomizer.class.getName())
+                .build()) {
+            classCreator.addAnnotation(Singleton.class);
+
+            try (MethodCreator customize = classCreator.getMethodCreator("customize", void.class, ObjectMapper.class)) {
+                ResultHandle objectMapper = customize.getMethodParam(0);
+
+                for (JacksonModuleBuildItem jacksonModule : jacksonModules) {
+                    if (jacksonModule.getItems().isEmpty()) {
+                        continue;
+                    }
+
+                    /*
+                     * Create code similar to the following:
+                     *
+                     * SimpleModule module = new SimpleModule("somename");
+                     * module.addSerializer(Foo.class, new FooSerializer());
+                     * module.addDeserializer(Foo.class, new FooDeserializer());
+                     * objectMapper.registerModule(module);
+                     */
+                    ResultHandle module = customize.newInstance(
+                            MethodDescriptor.ofConstructor(SimpleModule.class, String.class),
+                            customize.load(jacksonModule.getName()));
+
+                    for (JacksonModuleBuildItem.Item item : jacksonModule.getItems()) {
```

Comment:
Why don't you have a serializer list and then a deserializer list in your module?

That would simplify the code a lot and be more readable. I might be missing something obvious though.

Reply:
In Jackson you need to specify the class that the serializer and deserializer are "targeting". So with `JacksonModuleBuildItem.Item` the targetClass and serializer/deserializer are grouped together.

Basically what the loop generates is something like:

```
SimpleModule module = new SimpleModule("somename");
module.addSerializer(Foo.class, new FooSerializer());
module.addDeserializer(Foo.class, new FooDeserializer());
objectMapper.registerModule(module);
```

Does it make more sense now?